### PR TITLE
chore: Hsn fix for 2c+

### PR DESCRIPTION
**Pull request by jsturtevant**
**Reason for Change**:

Hsn fix for 2c+

If multiple LB policies are included for same endpoint then HNS hangs. this fix forces an error  so restarts to HNS are not required.

**Issue Fixed**:

**Credit Where Due**:
https://github.com/Azure/AgentBaker/pull/628
/cc @AbelHu @mainred

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by AbelHu — APPROVED** [review]

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -413,6 +413,11 @@ try
         Write-Log "Disable Internet Explorer compat mode and set homepage"
         Set-Explorer
 
+        # if multple LB policies are included for same endpoint then HNS hangs.
+        # this fix forces an error  
+        Write-Host "Enable a HNS fix in 2021-2C+"
```
Comment (AbelHu) [inline parts/k8s/kuberneteswindowssetup.ps1:418]: @jsturtevant do you mean that the code in configure-windows-vhd.ps1 does not work?

Comment (jsturtevant) [reply parts/k8s/kuberneteswindowssetup.ps1:418]: vhd should work, but we also have cases where VHD is not required in aks-engine so wanted to make sure it was enabled for those

**Review by AbelHu — APPROVED** [review]

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 2]
```diff
@@ -413,6 +413,11 @@ try
         Write-Log "Disable Internet Explorer compat mode and set homepage"
         Set-Explorer
 
+        # if multple LB policies are included for same endpoint then HNS hangs.
+        # this fix forces an error  
+        Write-Host "Enable a HNS fix in 2021-2C+"
+        reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\hns\State /v HNSControlFlag /t REG_DWORD /d 1
+
```
Comment (marosset) [inline parts/k8s/kuberneteswindowssetup.ps1:420]: Will this ever be set by default in some future OS updates or will we always need to set this? 
Is this only needed on Windows Server 2019 or all Windows Server OS versions?

File: pkg/engine/templates_generated.go [hunk 3]
```diff
@@ -17492,6 +17492,11 @@ try
         Write-Log "Disable Internet Explorer compat mode and set homepage"
         Set-Explorer
 
+        # if multple LB policies are included for same endpoint then HNS hangs.
+        # this fix forces an error  
+        Write-Host "Enable a HNS fix in 2021-2C+"
+        reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\hns\State /v HNSControlFlag /t REG_DWORD /d 1
```
Comment (marosset) [inline pkg/engine/templates_generated.go:17498]: nit: Can you use `Set-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Services\hns\State' -Name HNSControlFlag -Value 1 -PropertyType DWORD`

File: parts/k8s/kuberneteswindowssetup.ps1 (hunk 2 above)
Comment (jsturtevant) [reply parts/k8s/kuberneteswindowssetup.ps1:420]: @daschott do you have insight here?

**Review by jackfrancis — APPROVED** [review]
code lgtm

**Review by AbelHu — APPROVED** [review]

**Review by marosset — APPROVED** [review]
/lgtm

**Pull request merged**